core/templates/components/skill-selector/skill-selector.directive.ts

```diff
@@ -37,23 +39,133 @@ angular.module('oppia').directive('selectSkill', [
         function(
             $scope, $uibModal, $rootScope) {
           var ctrl = this;
-          $scope.selectSkill = function(skillId) {
-            $scope.selectedSkillId = skillId;
-          };
           ctrl.$onInit = function() {
-            $scope.sortedSkillSummaries = $scope.getSortedSkillSummaries();
-            $scope.skillSummariesInitial = [];
-            $scope.skillSummariesFinal = [];
+            $scope.selectedSkill = null;
+            $scope.categorizedSkills = $scope.getCategorizedSkills();
+            $scope.allowSkillsFromOtherTopics =
+              $scope.canAllowSkillsFromOtherTopics();
+            $scope.checkIfEmpty = function(skills) {
+              return (skills.length === 0);
+            };
+            $scope.skillChanged = function() {
+              $scope.selectedSkillId = $scope.selectedSkill;
+            };
+            $scope.isUntriagedSkills = function(topicName) {
```

Comment:
done.
We need to check for the key `untriaged_skills` in `categorizedSkillsDict`, because the structure of other keys and `untriaged_skills` is different and we need to handle them separately.